DependencyInjection/DoctrineExtension.php

```diff
@@ -196,6 +202,14 @@ protected function dbalLoad(array $config, ContainerBuilder $container)
         $connWithLogging   = [];
         $connWithProfiling = [];
         $connWithBacktrace = [];
+        $timingByConnection = [];
+        $skipTiming = false;
+
+        if (! $container->hasParameter('kernel.runtime_mode') && ! $container->hasParameter('kernel.runtime_mode.worker')) {
```

Comment:
checking the runtime mode at compile time looks suspicious to me. Those parameters are meant to have their values checked at runtime.

Reply:
Oh, ok, should we keep the listener in all cases, even in no worker mode? thanks for your feedback!